core/src/main/java/org/aerogear/mobile/core/MobileCore.java

```diff
@@ -76,73 +79,69 @@ private MobileCore(final Context context, final Options options)
 
-        // -- Setting default http layer
-        if (options.httpServiceModule == null) {
-            OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        OkHttpCertificatePinningParser certificatePinning =
+                        new OkHttpCertificatePinningParser(httpsConfig.getCertPinningConfig());
+        builder.certificatePinner(certificatePinning.parse());
 
-            OkHttpCertificatePinningParser certificatePinning =
-                            new OkHttpCertificatePinningParser(httpsConfig.getCertPinningConfig());
-            builder.certificatePinner(certificatePinning.parse());
+        builder.connectTimeout(DEFAULT_CONNECT_TIMEOUT, TimeUnit.SECONDS)
+                        .writeTimeout(DEFAULT_WRITE_TIMEOUT, TimeUnit.SECONDS)
+                        .readTimeout(DEFAULT_READ_TIMEOUT, TimeUnit.SECONDS);
+        final OkHttpServiceModule httpServiceModule = new OkHttpServiceModule(builder.build());
+        ServiceConfiguration configuration = this.servicesConfig.get(httpServiceModule.type());
+        if (configuration == null) {
+            configuration = new ServiceConfiguration.Builder().build();
+        }
 
-            builder.connectTimeout(DEFAULT_CONNECT_TIMEOUT, TimeUnit.SECONDS)
-                            .writeTimeout(DEFAULT_WRITE_TIMEOUT, TimeUnit.SECONDS)
-                            .readTimeout(DEFAULT_READ_TIMEOUT, TimeUnit.SECONDS);
-            final OkHttpServiceModule httpServiceModule = new OkHttpServiceModule(builder.build());
-            ServiceConfiguration configuration = this.servicesConfig.get(httpServiceModule.type());
-            if (configuration == null) {
-                configuration = new ServiceConfiguration.Builder().build();
-            }
+        httpServiceModule.configure(this, configuration);
 
-            httpServiceModule.configure(this, configuration);
+        this.httpLayer = httpServiceModule;
 
-            this.httpLayer = httpServiceModule;
-        } else {
-            this.httpLayer = options.httpServiceModule;
-        }
     }
 
     /**
      * Initialize the AeroGear system
      *
      * @param context Application context
-     * @return MobileCore instance
      */
-    public static MobileCore init(final Context context) throws InitializationException {
-        return init(context, new Options());
+    public static void init(final Context context) throws InitializationException {
+        init(context, new Options());
     }
 
     /**
      * Initialize the AeroGear system
      *
      * @param context Application context
      * @param options AeroGear initialization options
-     * @return MobileCore instance
      */
-    public static MobileCore init(final Context context, final Options options)
+    public static void init(final Context context, final Options options)
                     throws InitializationException {
-        return new MobileCore(context, options);
+        instance = new MobileCore(context, options);
```

Comment:
I'm not 100% sure about it. The library will create an instance automatically but the user/dev should wanna replace some default options and re-create it. 

If we decide to prevent recreation we should provide some way to change user/dev replace/change default options. wdyt?

Reply:
right, I get it now. Personally I don't like the idea to ask developers to re-init the who object just to reset the options. I think the MobileCore itself should provide a method to allow set the options after it's being created. This way the instance of the MobileCore will only need to be inited once.

Reply:
+9001 I will add that and make it immutable.